File: trunk/AI/RAI/java-cgdk/src/main/java/model/Player.java
package model;

/**
 * Содержит данные о текущем состоянии игрока.
 */
public final class Player {
    private final String name;
    private final int score;
    private final boolean strategyCrashed;

    public Player(String name, int score, boolean strategyCrashed) {
        this.name = name;
        this.score = score;
        this.strategyCrashed = strategyCrashed;
    }

    /**
     * @return Возвращает имя игрока.
     */
    public String getName() {
        return name;
    }

    /**
     * @return Возвращает текущее количество баллов, набранных игроком.
     */
    public int getScore() {
        return score;
    }

    /**
     * @return Возвращает специальный флаг --- показатель того, что стратегия игрока <<упала>>.
     *         Более подробную информацию можно найти в документации к игре.
     */
    public boolean isStrategyCrashed() {
        return strategyCrashed;
    }
}

File: AI/RAI/cpp-cgdk/model/Player.cpp
#include "Player.h"

using namespace model;
using namespace std;

Player::Player(string name, int score, bool strategy_crashed)
: name_(name), score_(score), strategy_crashed_(strategy_crashed) { }

string Player::name() const {
	return name_;
}

int Player::score() const {
	return score_;
}

bool Player::strategy_crashed() const {
	return strategy_crashed_;
}
